### PR TITLE
ADD: Feature/file query

### LINKED ISSUE #85: Implement FileQuery
This is a wrapper class for file search queries.

### PR DESCRIPTION
**Pull request by Segelzwerg**
closes #85

### CODE REVIEW

File: src/test/java/mops/businesslogic/FileQueryImplTest.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    }
+
+    @Test
+    public void noMatchingTagTest() {
+        FileQuery fileQuery = FileQuery.builder()
+                .tags(List.of("lösungen"))
+                .build();
+        assertThat(fileQuery.checkMatch(fileInfo)).isFalse();
+    }
+}
```
Comment (iTitus) [inline src/test/java/mops/businesslogic/FileQueryImplTest.java]: missing new line

File: src/main/java/mops/businesslogic/FileQueryImpl.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import mops.persistence.file.FileInfo;
+
+import java.util.List;
+import java.util.function.Function;
+
+@SuppressWarnings("PMD.LawOfDemeter") //these are a streams or database objects
+@AllArgsConstructor(access = AccessLevel.PACKAGE)
+public class FileQueryImpl implements FileQuery {
```
Comment (iTitus) [inline src/main/java/mops/businesslogic/FileQueryImpl.java]: can you replace the FileQuery interface with this implementation?

Comment (Segelzwerg) [reply src/main/java/mops/businesslogic/FileQueryImpl.java]: sure

File: src/main/java/mops/businesslogic/FileQueryBuilder.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+     */
+    private List<String> fileNames;
+    /**
+     * File types to search for.
+     */
+    private List<String> types;
+    /**
+     * File tags to search for.
+     */
+    private List<String> tags;
```
Comment (iTitus) [inline src/main/java/mops/businesslogic/FileQueryBuilder.java]: look at the FileInfo builder to see how collections are built

Comment (iTitus) [reply src/main/java/mops/businesslogic/FileQueryBuilder.java]: you don't set them, you add to existing ones

Comment (Segelzwerg) [reply src/main/java/mops/businesslogic/FileQueryBuilder.java]: ok I will try that

File: src/main/java/mops/businesslogic/FileQuery.java [hunk 4]
```diff
@@ -1,16 +1,70 @@
...
+     * List of file types to search for.
+     */
+    private List<String> types;
+
+    /**
+     * List of file tags to search for.
+     */
+    private List<String> tags;
+
+
```
Comment (iTitus) [inline src/main/java/mops/businesslogic/FileQuery.java]:
```suggestion

```
oh im seeing double again

File: src/main/java/mops/businesslogic/FileQueryBuilder.java [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+    public FileQueryBuilder owner(@NonNull String owner) {
+        if (owner.isEmpty()) {
+            throw new IllegalArgumentException("Owner must not be empty.");
+        }
+        owners.add(owner);
+        return this;
+
+    }
+
+
```
Comment (iTitus) [inline src/main/java/mops/businesslogic/FileQueryBuilder.java]:
```suggestion

```

File: src/main/java/mops/businesslogic/FileQuery.java [hunk 6]
```diff
@@ -1,16 +1,68 @@
...
 import mops.persistence.file.FileInfo;
 
-/**
- * Object that contains query parameter for searching files in a group.
- */
-public interface FileQuery {
+import java.util.List;
+import java.util.function.Function;
+
+@SuppressWarnings("PMD.LawOfDemeter") //these are a streams or database objects
```
Comment (iTitus) [inline src/main/java/mops/businesslogic/FileQuery.java]:
```suggestion
@SuppressWarnings("PMD.LawOfDemeter") //these are streams or builders
```

File: src/main/java/mops/businesslogic/FileQueryBuilder.java [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+import java.util.List;
+
+@NoArgsConstructor(access = AccessLevel.PACKAGE)
+@SuppressWarnings("PMD.AvoidFieldNameMatchingMethodName")
+public class FileQueryBuilder {
+    /**
+     * List of owner to search for.
+     */
+    private List<String> owners = new ArrayList<>();
+
```
Comment (iTitus) [inline src/main/java/mops/businesslogic/FileQueryBuilder.java]:
```suggestion
```

File: src/main/java/mops/businesslogic/FileQueryBuilder.java [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+        return this;
+    }
+
+    /**
+     * @param fileName new file name to search for
+     * @return this
+     */
+    public FileQueryBuilder fileName(@NonNull String fileName) {
+        if (fileName.isEmpty()) {
+            throw new IllegalArgumentException("File name must be empty.");
```
Comment (iTitus) [inline src/main/java/mops/businesslogic/FileQueryBuilder.java]:
```suggestion
            throw new IllegalArgumentException("File name must not be empty.");
```

**Pull request merged**